### ISSUE TITLE
active_pane.pane_id returns a different value to WEZTERM_PANE

### ISSUE THREAD

**Issue by loralb** [wez/wezterm#4187]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

No, and I'll explain why below

### Describe the bug

I defined `wezterm.on(format-tab-title, ...)` to add the `pane_id` of the active pane in the tab title; when connecting to a wezterm multiplexer though (with `wezterm connect <DOMAIN_NAME>`, in my case the domain is an SSH domain, not sure if that makes a difference), the value printed in the tab title is different from the value stored in the `WEZTERM_PANE` environment variable.
Specifically, I tried to start from scratch (by killing the wezterm on the remote machine) and noticed that there is a mismatch starting already from the very first pane, precisely an off-by-one mismatch, i.e. `echo $WEZTERM_PANE` returns 0, the value printed in the tab title returns 1. The mismatch is the same for all subsequent panes I created, but before starting from scratch the mismatch did not look to have any discernable pattern (for example, `WEZTERM_PANE` = 22, pane_id = 37). TBH, before killing the wezterm instance on the remote machine, I had a session with a few workspaces, each with a few tabs and panes, so it was a fairly complex setting.

About why I did not try latest nightly build, I looked at the commits since last release and did nof find any mention of `pane_id` nor `WEZTERM_PANE` + changing the wezterm version on the remote machine is not trivial.

I realise that this information may not be enough to pin down the problem, but I honestly have no idea what else to add; please, ping me asking whatever you need me to do to help solving this.

P.S.: the problem does not look to happen in a local wezterm instance.
P.P.S.: this happens by connecting to a remote linux server from a linux machine, but my (very wild) guess is that the problem is platform independent.

### Configuration

```
wezterm.on('format-tab-title', function(tab)
	local index = tab.tab_index + 1
	local common = ' #' .. index .. ': '
	local title = tab.tab_title
	if not title or #title == 0 then
		title = tab.active_pane.title
	end

	return common .. title .. ' [pane_id: ' .. tab.active_pane.pane_id .. ']'
end)
```

**Comment by loralb** [comment]
Makes perfect sense, thanks for explaining!

So, I guess there is not an easy way to get the remote `pane_id` other than `echo $WEZTERM_PANE`?

To explain why I'd like that, let me make an example: I start a `gdb` session and then I'd like to know the `pane_id` where it is running (for example, to use it in another tab where I have my editor open, in order to `wezterm cli send-text --pane-id [pane_id] [open_file_path]:[open_file_line_number]`); the only way that comes up in my mind is to `CTRL-z` the `gdb` session in order to `echo` the `WEZTERM_PANE` variable and then `fg`.
I mean, it works, but it is a bit cumbersome (to the point that copy-pasting the path from the editor to the gdb session becomes easier).

**Comment by wez** [comment]
Sounds like you want:
* https://github.com/wez/wezterm/issues/4147

**Comment by loralb** [comment]
I'll look into it, thank you!

**Comment by wez** [comment]
Closing this as a duplicate of #4147
